ui/repetition.reel/repetition.js

```diff
@@ -700,6 +719,62 @@ var Repetition = exports.Repetition = Component.specialize(/** @lends Repetition
     // Implementation:
     // ----
 
+    _selection: {
+        value: null
+    },
+
+    _cancelSelectionRangeChangeListener: {
+        value: null
+    },
+
+    selection: {
+        get: function () {
+            return this._selection;
+        },
+        set: function (value) {
+            if (this._selection !== value) {
+                this._selection = value;
+                if (this.contentController && this.contentController.selection !== value) {
+                    this.contentController.selection = value;
+                }
+                if (this._cancelSelectionRangeChangeListener) {
+                    this._cancelSelectionRangeChangeListener();
+                }
+                if (value) {
+                    this._cancelSelectionRangeChangeListener = (
+                        value.addRangeChangeListener(this, "selection")
+                    );
+                    this.handleSelectionRangeChange(value, []);
+                } else {
+                    this._cancelSelectionRangeChangeListener = null;
+                }
+            }
+        }
+    },
+
+    handleSelectionRangeChange: {
```

Comment:
In a multi-selection scenario the loop would end being in worst case O(n ^ 2). Suppose we have a million iterations and we perform a select all function... that would be very slow.

So I'm using the Map as a hash, and I assume the hash look up might range depending on the implementation between O(1) and O(log n), so in the worst case scenario - select all - I would end with a maximum of O(n log n).

As the most common case for selection is not multi-selection, what I propose is to add a conditional in the code for simple selection, if add and remove lengths are lower or equal to 1, then I could just loop and then I would avoid the Map creation in that case. Does it sound good?